File: app/src/main/java/com/viginfotech/chennaitimes/util/NetworkUtil.java
package com.viginfotech.chennaitimes.util;

import android.content.Context;
import android.net.ConnectivityManager;

/**
 * Created by anand on 3/10/16.
 */
public class NetworkUtil {

    public static boolean isOnline(Context context){
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
       return connectivityManager.getActiveNetworkInfo() != null &&
               connectivityManager.getActiveNetworkInfo().isConnectedOrConnecting();

    }
}

File: app/src/main/java/com/viginfotech/chennaitimes/util/NetworkUtil.kt
package com.viginfotech.chennaitimes.util

import android.content.Context
import android.net.ConnectivityManager

/**
 * Created by anand on 3/10/16.
 */
object NetworkUtil {

    fun isOnline(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return connectivityManager.activeNetworkInfo != null && connectivityManager.activeNetworkInfo.isConnectedOrConnecting

    }
}
